### ISSUE TITLE
protected-access false positive with Generic classes

### ISSUE THREAD

**Issue by Aegdesil** [pylint-dev/pylint#8600]
### Bug description

When linting this code, I get a protected-access error which should not happen.
Removing the Generics fixes the issue.

```
from typing import Generic, TypeVar

T = TypeVar("T")


class Parent(Generic[T]):
    def _foo(self):
        pass


class Child(Parent[T]):
    def _foo(self):
        Parent._foo(self)
```

### Command used

```shell
pylint test.py
```

### Pylint output

```shell
[...]
test.py:13:8: W0212: Access to a protected member _foo of a client class (protected-access)
[...]
```

### Expected behavior

No error from pylint.

### Pylint version

```shell
pylint 2.17.2
astroid 2.15.3
Python 3.10.0 (default, Oct 26 2021, 11:33:05) [Clang 13.0.0 (clang-1300.0.29.3)]
```

### OS / Environment

Mac OS
